src/Compilers/CSharp/Portable/FlowAnalysis/PreciseAbstractFlowPass_Switch.cs

```diff
@@ -182,11 +188,17 @@ private void VisitDecisionTree(DecisionTree decisionTree)
                 case DecisionTree.DecisionKind.ByValue:
                     {
                         var byValue = (DecisionTree.ByValue)decisionTree;
+                        var expressionIsConstant = byValue.Expression.ConstantValue != null;
+                        bool valueHandled = false;
                         foreach (var kvp in byValue.ValueAndDecision)
                         {
-                            VisitDecisionTree(kvp.Value);
+                            if (!expressionIsConstant || Equals(byValue.Expression.ConstantValue.Value, kvp.Key))
```

Comment:
OK, sure.

---
In reply to: [69162180](https://github.com/dotnet/roslyn/pull/12250#discussion_r69162180) [](ancestors = 69162180)